src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/UpdateCertificateWithAmendReasonCommandHandlerTests/UpdateCertificateWithAmendReasonCommandHandlerTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.Extensions.Logging;
+using Moq;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using SFA.DAS.AssessorService.Api.Types.Enums;
+using SFA.DAS.AssessorService.Api.Types.Models.Certificates;
+using SFA.DAS.AssessorService.Application.Handlers.Certificates;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Consts;
+using SFA.DAS.AssessorService.Domain.Entities;
+using SFA.DAS.AssessorService.Domain.JsonData;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Application.UnitTests.Handlers.Certificates.UpdateCertificateWithAmendReasonCommandHandlerTests
+{
+    public class UpdateCertificateWithAmendReasonCommandHandlerTests
+    {
+        [Test]
```

Comment:
should there be a test for when GetCertificate() returns null and we throw NotFoundException?

Reply:
Added additional tests